zc_common/remote_resource/tests.py

```diff
@@ -12,11 +12,54 @@ class ResponseTestCase(APITestCase):
     FAILURE_HIGH_LEVEL_KEYS = ['errors']
     FAILURE_DATA_KEYS = ['status', 'source', 'detail']
 
+    INSTANCE_TOP_LEVEL_KEYS = ["type", "id", "attributes"]
+
     def convert_to_list(self, data):
-        if isinstance(data, list):
+        if isinstance(data, dict):
+            return [data]
+        elif isinstance(data, list):
             return data
         else:
-            return [data]
+            return data.all()
+
+    def validate_instance_in_data(self, data, instance, attributes, relationship_keys=None):
+        self.assertTrue(all(key in data for key in self.INSTANCE_TOP_LEVEL_KEYS))
+
+        instance_attributes = data["attributes"]
+        for key in attributes:
+            self.assertEqual(instance_attributes[camelize(key, False)], getattr(instance, key))
+
+        if not relationship_keys:
+            return True
```

Comment:
This is really personal opinion, I'm a supporter of early returns in these cases:
https://en.wikibooks.org/wiki/Computer_Programming/Coding_Style/Minimize_nesting